File: TypeScript/contains_duplicate/1.ts
function containsDuplicate(nums: number[]): boolean {
  const set = new Set();

  for (let n = 0; n < nums.length; n++) {
    if (set.has(nums[n])) return true;
    else set.add(nums[n]);
  }
  return false;
}

console.log(containsDuplicate([1, 2, 3, 4]));

File: src/Python/contains_duplicate/1.py
# Crea un algoritmo para encontrar duplicados en un array de números
# Parámetro: nums, array de números
# Retorno: Booleano, True si hay duplicados, False si no hay duplicados

def contains_duplicate(nums):
    uniques = set()
    for number in nums:
        if number in uniques:
            return True
        uniques.add(number)
    return False


print(contains_duplicate([1, 2, 3, 4, 2]))
